fml/thread_local.cc

```diff
@@ -25,7 +25,12 @@ void* ThreadLocalPointer::get() const {
 
 void* ThreadLocalPointer::swap(void* ptr) {
   void* old_ptr = get();
-  FML_CHECK(pthread_setspecific(key_, ptr) == 0);
+  int err = pthread_setspecific(key_, ptr);
+  if (err) {
+    FML_LOG(ERROR) << "pthread_setspecific failed (" << err
```

Comment:
You can send a stream to `FML_CHECK(false)` instead of logging first and then tripping the assertion.